### PR TITLE
Add checks for labels and input sizes for split function

### PR DESCRIPTION
**Pull request by swaingotnochill**
Issue: #2991 

Document discussing the changes ideas:
https://docs.google.com/document/d/1fdNIKlXlcOAXTVIhdSeDvxYS9F5-95qviFZxjsBswBQ/edit?usp=sharing

@zoq

### CODE REVIEW

File: src/mlpack/tests/split_data_test.cpp [hunk 1]
```diff
@@ -175,6 +175,20 @@ TEST_CASE("SplitLabeledDataResultMat", "[SplitDataTest]")
   CheckDuplication(std::get<2>(value), std::get<3>(value));
 }
 
+TEST_CASE("SplitCheckSize", "[SplitDataTest]")
+{
+  arma::mat input(2,10);
+  input.randu();
```
Comment (zoq) [inline src/mlpack/tests/split_data_test.cpp:181]:
```suggestion
  arma::mat input = randu<arma::mat>(2, 10);
```

In one pass.

File: src/mlpack/tests/split_data_test.cpp [hunk 2]
```diff
@@ -175,6 +175,20 @@ TEST_CASE("SplitLabeledDataResultMat", "[SplitDataTest]")
...
+
+  const Row<size_t> firstLabels = arma::linspace<Row<size_t>> (0, input.n_cols-1, input.n_cols);
+
+  const Row<size_t> secondLabels = arma::linspace<Row<size_t>> (0, input.n_cols, input.n_cols+1);
```
Comment (zoq) [inline src/mlpack/tests/split_data_test.cpp:185]:
```suggestion
  const Row<size_t> firstLabels = arma::linspace<Row<size_t>>(0, input.n_cols - 1, input.n_cols);

  const Row<size_t> secondLabels = arma::linspace<Row<size_t>>(0, input.n_cols, input.n_cols + 1);
```

Space between operators.

**Review by zoq — COMMENTED** [review]
Some minor comments.

**Comment by RishabhGarg108** [comment]
Hey @swaingotnochill, I saw that the `SplitMatrixLabeledData` test is failing in the build. The reason for this is that the `labels` used in that test is a matrix. I also encountered this case when working on xgboost and I made [this](https://github.com/mlpack/mlpack/pull/3025/commits/82d57e0b9a478e07e701c3a63c9eb879dcf43433) fix. You can also do this to resolve the error. Also, remember to update the error message in the function too. I forgot to do it there. 👍

**Comment by swaingotnochill** [comment]
> Hey @swaingotnochill, I saw that the `SplitMatrixLabeledData` test is failing in the build. The reason for this is that the `labels` used in that test is a matrix. I also encountered this case when working on xgboost and I made [this](https://github.com/mlpack/mlpack/pull/3025/commits/82d57e0b9a478e07e701c3a63c9eb879dcf43433) fix. You can also do this to resolve the error. Also, remember to update the error message in the function too. I forgot to do it there.

Yaa I saw that a single test case failed and this was coz it was using matrix label datatype. It was helpful.Thanks

**Review by zoq — APPROVED** [review]
Looks good to me, no further comments from my side.

**Comment by RishabhGarg108** [comment]
@zoq, I guess we can merge this one now :)

**Pull request merged**